### DISCUSSION TITLE
“mix-tape” grouping (different years inside same album)

### DISCUSSION THREAD

**Discussion by lzbk** [ampache/ampache#3449] [General]
I have found many mix-tapes with tracks having the year of the original release rather than that of the mix (which I actually like) and ampache treats them as separate albums (which is understandable, in that different bands might give the same name to an album years apart). Maybe the tracks could be kept together when the "album artist" and "album name" tags are consistent over the different tracks from different years and authors ? (especially when they are in the same directory)

**Comment by lachlan-00** [comment]
Do you have any examples? The mix tape is still released on a single date

**Comment by lzbk** [comment]
Thanks for your answer and sorry for the late reply.

I might not have phrased it correctly, I'm not speaking of [hip-hop (“professional”) mix-tapes](https://www.datpiff.com/) (+ pros are not necessarily the most eager taggers btw), but of mix-tapes people send one another, i.e. compilations…

But the use of tags is by essence conventional. The “date” tag is no exception… For instance, if you include in your 2020 mix "[Dazed and confused](https://en.wikipedia.org/wiki/Dazed_and_Confused_(song)#Led_Zeppelin), you might tag it:
* 1969: the version of [Led Zeppelin I](https://en.wikipedia.org/wiki/Led_Zeppelin_(album))
* 1971: BBC session date of recording (April 1st 1971)
* 1997: If you got it off [BBC Session standard edition](https://en.wikipedia.org/wiki/BBC_Sessions_(Led_Zeppelin_album))
* 2016: If you got it off [BBC Session Complete Sessions edition](https://en.wikipedia.org/wiki/BBC_Sessions_(Led_Zeppelin_album))
* 2020: Date of your mix

In communities where people create mixes for other users, it is common practice (even though not necessarily a norm) to chose to keep the year of the original release date of the song (or its recording date) in the “year” tag, rather than that of the mix. Precisely to give an idea to the listener of when in time it was, what version was chosen, etc.

**e.g.**

![mixed years](https://user-images.githubusercontent.com/3218235/74849777-6753ae00-5339-11ea-8656-1563323fb94a.png)

And now here is the compilation viewed from Ampache

![Funky Money](https://user-images.githubusercontent.com/3218235/74852079-696b3c00-533c-11ea-9d8e-403430075f0f.png)

Hence my suggestion:
>Maybe the tracks could be kept together when the "album artist" and "album name" tags are consistent over the different tracks from different years and authors ? (especially when they are in the same directory)

**Comment by SimonHova** [comment]
Perhaps I can fill in some missing information. Ampache uses [Musicbrainz](https://musicbrainz.org/) as it's cataloging system, and that's where the mbid (Musicbrainz ID) comes from. With a few exceptions, they would not issue a MBID for your downloaded mixtape, and that's where the album year (date compiled) would come from.

However, each one of the recordings that you'd listed has an MBID for it, from the original 1969 [version](https://musicbrainz.org/recording/553510ce-68dd-4e85-9e8c-eafb23d43bab), to the 1971 live [version](https://musicbrainz.org/recording/f7422a6c-5b2f-4a39-81b1-6023732e1557).

For  MusicBrainz to properly tag the recordings, I'd recommend cataloging the tracks to their original albums, then making an Ampache playlist for the tracks you wanted to include. You can modify the date, and change any other metadata that way.

**Comment by lzbk** [comment]
@SimonHova I think you are mistaken here. 

Ampache resorts to Musicbrainz, only when lacking information… or at least that is what it is supposed to do in my case :

> ```metadata_order = "getID3,filename,MusicBrainz,TheAudioDb"```

In this case, it actually does exactly that, since [the 1982 version of D.M.S.R.](https://musicbrainz.org/search?query=D.M.S.R&type=recording&method=indexed) is filed under the funky monkey compilation and not under [Sacrifice](https://musicbrainz.org/release/9dade22b-2733-43bd-a342-524017c70afc). Additionnally, if I decide to lose the info of the chronology and file everything under the same year, it will work fine (despite [1997](https://musicbrainz.org/recording/f43b5ee2-f918-4c8a-862f-9f051d44e503), [1998](https://musicbrainz.org/recording/3db9f20b-e88a-4e9b-8156-262f7220767e), [2001](https://musicbrainz.org/recording/f43b5ee2-f918-4c8a-862f-9f051d44e503), etc. tracks) :
![Screenshot_2020-02-28](https://user-images.githubusercontent.com/3218235/75564608-be9a0280-5a4c-11ea-87ae-8165962e80ef.png)

The issue is not what metadata is stored, rather how they are handled…

A decision needed to be taken when two tracks shared the same album name, with different years and artist. And the decision was that they are different albums. It has not always been the case as ampache used to file [The W](https://musicbrainz.org/release/f6fa5dfe-b939-3c49-ad26-3adc7315acd5) and [W](https://musicbrainz.org/release/539380c7-e7b2-4b27-8eb7-fe40b34552cf) as the same album (≠ year ≠ artist). I ended up erasing planningtorock… We do not want to go back to that.

The problem is that an album requires a year (as part of a multiple key):
![album structure](https://user-images.githubusercontent.com/3218235/75566210-98c22d00-5a4f-11ea-908e-98a5a0dd6b7d.png)

But I still think, there should be room for allowing keeping track of the original recording when getting a compilation. The information is valuable to the listener, see the info made available on the Pixies' [complete b-side for instance](https://musicbrainz.org/release-group/b9f32dc3-7116-3f63-9ad6-84e56813e87a).

After digging into the DB, the system actually allows to do what I want (and what would seem the most logical to me): an album year different from each track's year :
![image](https://user-images.githubusercontent.com/3218235/75569740-feb1b300-5a55-11ea-85dc-8247595f9294.png)

The problems are:
1. I modified it manually in the db, with phantom (empty) records and possible inconsistencies (+ whenever I clean up the catalog, I'll lose all my work)
1. If I am to do it cleanly, through ampache, when you modify an album's year it modifies the year of every single track (That means that I have to manually change the year of each “sub-album” then manually revert the year of each song within the modified album, which is very tedious)
2. The display of the track info on the track page is inconsistent (the album year is OK from the album page, but not from the track page, where the album year is that of the track).
![image](https://user-images.githubusercontent.com/3218235/75567148-53066400-5a51-11ea-8665-3efe192ccddc.png)
3. It is not possible to modify a track's metadata from ampache without adding a license.

Problem 2 does not seem too complicated. For problem 3, the license should start with an empty field.

Problem 1 could be solved by adding another combo box, next to the year tag (or with the other combo boxes) saying "Do not alter track year" or something of the sort… The problem remains concerning automatic indexation. In this case, I stick with my suggestion, refined as :
>Same album name, same folder (or same album artist) are kept in the same album, the year of which should be, depending on availability
>1. an "album year" tag (in vorbis tags)
>1. any 4 digit number between parenthesis of the folder name
>1. year of the album on music brains (if the album name exists)
>1. the highest year among track years

**Comment by josu113** [comment]
I propose a solution to store the year of the songs in the compilations.
The database table 'songs' has a field 'year' that can be used for that

When you create or update a catalog, if a song has a release_type equals to "Compilation", the tag 'original_year' of the song can be stored in the field 'year' of the table 'songs' and ignore the field 'original_year' in the table 'albums'

Another solution to populate this field ('year' in table 'songs') could be use the frame TDRC (Recording time) of ID3v2.4, and the frame TDRL (Release time) for the field 'year' in table 'albums'
